lambda_handler.py

```diff
@@ -1,110 +1,110 @@
-    def save_kinesis_record(kinesis_client, record):
-        try:
-            kinesis_client.put_records(record)
-        except:
-            raise
+            for item in response['Items']:
+                destination_ddb.put_item(
+                    TableName="table2",
+                    Item=item
+                )
+
+        # This code uses a mutable default argument and modifies it to return. This would leak results across calls
+        def list_sns(sns, topics=[]):
+            response = sns.list_topics()
+            for topic_arn in response["Topics"]:
+                topics.append(topic_arn["TopicArn"])
+            return topics
+
+
+        # Infinite loop because a list is modified while being iterated over, Indices are not updated.
+        def infinite_loop():
+            words = ['aws', 'amazon', 'codeguru']
+            for w in words:
+                if len(w) > 4:
+                    words.insert(0, w)
+            return words
+
+        # Prefer DefaultDict over setDefult
+        def setdefault_example():
+            std_dict = dict()
+            for k, v  in enumerate(range(5)):
+                std_dict.setdefault(k, []).append(v)
+            return std_dict
+
+        # This method reads multiple file paths, open each file to load data, but forgets to close it, leading to resource leaks
+        # Further, it selectively processes content based on string find condition. The find() operation can be simply replaced
+        # with a membership tests because one does not need to know the position at which the search keyword appears.
+        def process_kinesis(kinesis, file_list_path):
+            flp = open(file_list_path, 'r')
+            for line in flp:
+                file_path = line.strip('\n').strip('\r\n')
+                fp = open(file_path, 'r')
+                for content in fp:
+                    if content.find("kinesis") != -1:
+                        record = load_kinesis_record(content)
+                        save_kinesis_record(kinesis, record)
+
+        # Do not call this function unless you're sure that the "cmd" is secure to run
+        # This function can be misused to carry out shell injection attacks.
+        # Further, the code is simply passing an exception, which is not the best practice
+        # Further, the code keeps track of successful loads by writing to a global variable; which can lead to inaccuracies in
+        # case of concurrent read/writes to the global variable.
+
+        def load_kinesis_record(cmd, mode='subprocess'):
+            global successes
+            kinesis_record = None
+            try:
+                if mode == "subprocess":
+                    kinesis_record = subprocess.call(cmd, shell=True)
+                else:
+                    kinesis_record = eval(cmd)
+                successes += 1
+            except Exception as e:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```